File: 0477-Total-Hamming-Distance/0447.py
class Solution:
  def totalHammingDistance(self, nums: List[int]) -> int:
    n = len(nums)
    ans = 0
    for i in range(30):
      c = sum(((val >> i) & 1) for val in nums)
      ans += c * (n - c)
    return ans

File: 401-500/0477-Total-Hamming-Distance/0447.java
class Solution {
  public int totalHammingDistance(int[] nums) {
    int ans = 0, n = nums.length;
    for (int i = 0; i < 30; ++i) {
      int c = 0;
      for (int val : nums) {
        c += (val >> i) & 1;
      }
      ans += c * (n - c);
    }
    return ans;
  }
}